highway.tex

```diff
@@ -24,807 +26,1323 @@
+%
+As a way of capturing and spreading the different validators' knowledge about the already existing messages, it adopts the DAG framework~\cite{moser1999byzantine,zamfir2018casper,baird2016hashgraph,GLSS19}, in which every message broadcast by a validator refers a certain set of  messages sent by validators before.
+%
+We will refer to such messages broadcast during normal protocol operation as \emph{units}, and the included references as \emph{citations}. More formally, each unit consists of the following data
+\begin{itemize}
+    \item {\bf Sender.} The index of the unit's sender (creator).
+    \item {\bf Citations.} A list of hashes of other units that the creator wants to attest to. 
+    \item {\bf Block.} In case a unit is produced by the validator appointed to produce a block at a given time, it is included in the unit. 
+\end{itemize}
 
-\subsection{Votes and Weights}
+All units to be considered correct must also a digital signature by its sender.
+%
+We denote the sender (creator) of a unit $u$ by $S(u)$.
+%
+As a given unit can only refer to previously constructed units, the citations contained in units can be seen as edges in a DAG (directed acyclic graph).
 
-The protocol will support a changing set of active validators with different weights, but we will study this aspect later. For now, we consider a fixed map:
 
-\begin{definition}
-  A \emph{validator map} is a map $\ww: \mathcal{V} \rightarrow \mathbb{R}_{\geq 0}$, assigning to each validator $v \in \mathcal{V}$ a \emph{weight} $\ww(v)$, such that at least one, but only finitely many validators are assigned a positive weight\footnote{To simplify the notation, instead of changing the set $\mathcal{V}$ itself, we will only change the map $w$, and assign $0$ to all validators who are not currently active. So, conceptually $\mathcal{V}$ is the set of all \emph{potential} validators and can even be infinite. In an implementation, $w$ should be represented as the finite map containing only the non-zero entries.}.
+In the protocol it will be often important not only whether a given unit $u$ cites directly some other unit $u'$, but whether there is a provable casual dependence between them, i.e., whether it can be proven that during the creation of $u$ its creator was aware of $u'$.
+%
+Such a notion is easily captured by the existence of the chain of citations connecting $u$ and $u'$.
+%
+In the presence of such a chain, we say that the unit $u$ is a  \emph{justification} of unit $u'$ or, in other words, that $u'$ justifies $u$.
+%
+As the justification relation is transitive, we interpret it as a partial order on the set of units and denote the fact that $u'$ justifies $u$ by $u'\leq u$.
+%
+We denote the set of all units strictly smaller than $u$ in this order, i.e., its \emph{downset}, by $D(u)$, and the set $D(u)\cup\{u\}$ as $\bar{D}(u)$.
```

Comment:
I guess [the technically correct terms](https://en.wikipedia.org/wiki/Upper_set#Upper_closure_and_lower_closure) would be "strict lower closure" (for `$D$`) and "lower closure" (for `$\bar{D}$`). But it's certainly _a_ downset, too, and "downset" is much shorter, so I'm fine with it. :+1: 
(@xcthulhu might not be, though? :grimacing: )

(On the other hand, our definition of "protocol state" now exactly matches the original meaning of "downset".)

Reply:
I think we should go with _strict lower closure_ and _lower closure_.  The lower closure of `S` is `$\downarrow S$`.  The order theory literature doesn't often refer to strict lower closures.  Perhaps we could write `v < u` rather than `v \in D(u)`?

Protocol states are downsets, that is the correct use of the term.

Reply:
Even besides the fact that we explicitly agreed to use the 'downset' on a call specifically devoted to the unification of terms, I find the proposed change problematic.;) We usually write $v < u$ rather than $v \in D(u)$ anyway, but such context is by far not the only one in which the notion of a downset is used - quite often we just want to say that the whole D(u) (or \bar{D}(u)) has some specific form, or is a subset of some other set, etc. 
We quite desperately need notation for what is now denoted as D(u) (which is much more prevalent in the paper than \bar{D}(u), although the second one is also used over 10 times), and unfortunately the traditional order theory does not provide it. Using \downarrow as a notation for _strict lower closure_ instead of a _lower closure_ would be probably much more misleading to people accustomed with the usual notation than what we use now. 
And while I know that usually downset means something different, intuitively the original meaning is not only not far off from the one adopted in the paper, but also shouldn't cause much confusion - the 'traditional' downset is a property of a set and as such does not accept any parameter. Anyone seeing D(u) (with parameter u) should instantly recognize that we are abusing the notation in some way.
If you have an idea how to solve that conundrum, we're still open to hear it (and change it). I'd like to make just a final note - changing the most crucial notation in the last stage of paper writing is a rather risky and errogenous practice, so I'd say that unless we find something that is clearly better, staying with the current choice is optimal from the perspective of paper quality.